hibernate-core/src/main/java/org/hibernate/type/descriptor/java/BooleanJavaType.java

```diff
@@ -193,10 +193,18 @@ public String getCheckCondition(String columnName, JdbcType jdbcType, BasicValue
 						(BasicValueConverter<Boolean, ?>) converter;
 				final Object falseValue = stringConverter.toRelationalValue( false );
 				final Object trueValue = stringConverter.toRelationalValue( true );
-				String[] values = new String[] {
-						falseValue != null ? falseValue.toString() : null,
-						trueValue != null ? trueValue.toString() : null
-				};
+				Object nullValue = null;
+				try {
+					nullValue =  stringConverter.toRelationalValue(null);
+				}
+				catch ( NullPointerException ignored ) {
+				}
+				final String[] values = new String[nullValue != null ? 3 : 2];
+				values[0] = falseValue != null ? falseValue.toString() : null;
+				values[1] = trueValue != null ? trueValue.toString() : null;
+				if ( nullValue != null ) {
+					values[2] =  nullValue.toString();
+				}
```

Comment:
Can we please hit these lines with Extract Method, the method is already too long.